src/indicators/flooding/floodscan/floodscan.py

```diff
@@ -32,46 +32,95 @@
         Args:
             country_name: name of the country of interest
             adm_level: admin level to compute the statistics on
+            custom_path: file path to the custom area shapefile
+            custom_id_col: the name of the id column for features in the custom area shapefile
+            custom_name: the name of the custom area (for the output file)
+            start_date: to filter by start date
+            end_date: to filter by end date
         """
-        config=Config()
+        config = Config()
         parameters = config.parameters(country_name)
         country_iso3 = parameters["iso3_code"]
-        adm_boundaries_path = os.path.join(DATA_DIR,PUBLIC_DATA_DIR, RAW_DATA_DIR,country_iso3,config.SHAPEFILE_DIR,parameters[f"path_admin{adm_level}_shp"])
-        ds=self.read_raw_dataset()
-        #get the affine transformation of the dataset. looks complicated, but haven't found better way to do it
-        coords_transform=ds.rio.set_spatial_dims(x_dim="lon",y_dim="lat").rio.write_crs("EPSG:4326").rio.transform()
-        #this takes a few hours to compute
-        df = self.compute_stats_per_area(ds,coords_transform,adm_boundaries_path,parameters[f"shp_adm{adm_level}c"])
-        self._write_to_processed_file(country_iso3,adm_level,df)
-
-    def compute_stats_per_area(self, ds, raster_transform, adm_path, adm_col, data_var="SFED_AREA",percentile_list=[2,4,6,8,10,20]):
+
+        ds = self.read_raw_dataset().sel(time=slice(start_date, end_date))
+
+        # get the affine transformation of the dataset. looks complicated, but haven't found better way to do it
+        coords_transform = (
+            ds.rio.set_spatial_dims(x_dim="lon", y_dim="lat")
+            .rio.write_crs("EPSG:4326")
+            .rio.transform()
+        )
+
+        if custom_path:
+            boundaries_path = custom_path
+
+            # this takes a few hours to compute
+            df = self.compute_stats_per_area(
+                ds, coords_transform, boundaries_path, custom_id_col,
+            )
+            self._write_to_processed_file(country_iso3, adm_level, df, custom_name)
+
+        else:
+            boundaries_path = os.path.join(
+                DATA_DIR,
+                RAW_DATA_DIR,
+                country_name,
+                config.SHAPEFILE_DIR,
+                parameters[f"path_admin{adm_level}_shp"],
+            )
+
+            # this takes a few hours to compute
+            df = self.compute_stats_per_area(
+                ds,
+                coords_transform,
+                boundaries_path,
+                parameters[f"shp_adm{adm_level}c"],
+            )
+            self._write_to_processed_file(country_iso3, adm_level, df, custom_name)
+
+    def compute_stats_per_area(
+        self,
+        ds,
+        raster_transform,
+        bound_path,
+        id_col,
+        data_var="SFED_AREA",
+        percentile_list=[2, 4, 6, 8, 10, 20],
```

Comment:
I know I did this, but I learned from Monica that you shouldn't set a list as default value😁 So set it here to None and then do an `if None: percentile_list=[2, 4, 6, 8, 10, 20]` (as I would argue these percentiles can be used across different projects, could also choose to not set any default value to it)